src/Orchard.Web/Modules/Lucene/Services/LuceneIndexProvider.cs

```diff
@@ -131,11 +133,17 @@ public class LuceneIndexProvider : IIndexProvider {
             Delete(indexName, indexDocuments.Select(i => i.ContentItemId));
 
             using (var writer = new IndexWriter(GetDirectory(indexName), _analyzerProvider.GetAnalyzer(indexName), false, IndexWriter.MaxFieldLength.UNLIMITED)) {
-                foreach (var indexDocument in indexDocuments) {
-                    var doc = CreateDocument(indexDocument);
-
-                    writer.AddDocument(doc);
-                    Logger.Debug("Document [{0}] indexed", indexDocument.ContentItemId);
+                var pageCount = (int)Math.Ceiling((decimal)indexDocuments.Count() / MAX_DOC_COUNT);
+                for (int i = 0; i < pageCount; i++)

```

Comment:
Please fix the formatting, K&R.
If you are using Visual Studio, you can install Rebracer which will do it automatically when loading Orchard.